specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/virtualWan.json

```diff
@@ -2634,6 +2634,231 @@
+          "vpnLinkConnections"
+        ],
+        "operationId": "VpnLinkConnections_Put_SharedKey",
+        "description": "Put the shared key of VpnLink connection specified.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "gatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the gateway."
+          },
+          {
+            "name": "connectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vpn connection."
+          },
+          {
+            "name": "linkConnectionName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the vpn link connection."
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ConnectionSharedKey"
+            },
+            "description": "Parameters supplied to the Begin Set vpn site link connection Shared key operation throughNetwork resource provider."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation set the shared key of VpnLink connection.",
+            "schema": {
+              "$ref": "#/definitions/ConnectionSharedKey"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation set shared key of VpnLink connection will complete asynchronously.",
+            "schema": {
+              "$ref": "#/definitions/ConnectionSharedKey"
+            }
+          },
+          "default": {
+            "description": "The operation set shared key of VpnLink connection could not be completed.",
+            "schema": {
+              "$ref": "./networkWatcher.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ResetVpnLinkConnection": {
+            "$ref": "./examples/VpnSiteLinkConnectionSharedKeySet.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/vpnGateways/{gatewayName}/vpnConnections/{connectionName}/vpnLinkConnections/{linkConnectionName}/sharedKey/reset": {
```

Comment:
>sharedKey/reset [](http://example.com/codeflow?start=198&length=15)

This is not right way to define a POST Action. Either it should be "/vpnLinkConnections/{linkConnectionName}/resetSharedKey" or "/vpnLinkConnections/{linkConnectionName}/sharedKeys/{sharedKeyName}/reset".

Reply:
> > sharedKey/reset
> 
> [ARMBlockingComment] This is not right way to define a POST Action. Either it should be "/vpnLinkConnections/{linkConnectionName}/resetSharedKey" or "/vpnLinkConnections/{linkConnectionName}/sharedKeys/{sharedKeyName}/reset".

We are following the API spec for Virtual Network Gateway Reset Shared Key for consistency. Please refer to the below link. It uses POST Action and "/connections/{virtualNetworkGatewayConnectionName}/sharedkey/reset".
[https://learn.microsoft.com/en-us/rest/api/network-gateway/virtual-network-gateway-connections/reset-shared-key?view=rest-network-gateway-2023-09-01&tabs=HTTP](url)
Example: POST https://management.azure.com/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/connections/{virtualNetworkGatewayConnectionName}/sharedkey/reset?api-version=2023-09-01